doc/src/challenges/XXXY-arc.md

```diff
@@ -0,0 +1,138 @@
+# Challenge XXXY[^challenge_id]: Verify atomically reference-counted Cell implementation
```

Comment:
Do you forsee any overlap/interaction with Challenge 7 (Safety of Methods for Atomic Types and `ReentrantLock`)?